### PR TITLE
Implement PHP RFC: Iterable

### PR DESCRIPTION
**Pull request by Hywan**
Link to RFC: https://wiki.php.net/rfc/iterable
Address https://github.com/tagua-vm/parser/issues/91.

### CODE REVIEW

File: spec/09-lexical-structure.md [hunk 1]
```diff
@@ -396,8 +396,8 @@ keyword:: one of
   'enddeclare' 'endfor' 'endforeach' 'endif' 'endswitch' 'endwhile' 'eval' 'exit'
   'extends' 'final' 'finally' 'for' 'foreach' 'function' 'global'
   'goto' 'if' 'implements' 'include' 'include_once' 'instanceof'
-  'insteadof' 'interface' 'isset' 'list' 'namespace' 'new' 'or' 'print' 'private'
-  'protected' 'public' 'require' 'require_once' 'return' 'static' 'switch'
+  'insteadof' 'interface' 'isset' 'iterable' 'list' 'namespace' 'new' 'or' 'print'
```
Comment (nikic) [inline spec/09-lexical-structure.md]: `iterable` is not a keyword, it's a reserved class name (similarly to how the scalar type hints are handled).

Comment (Hywan) [inline spec/09-lexical-structure.md]: Give me a minute to fix it :-).

Comment (Hywan) [inline spec/09-lexical-structure.md]: Done.

**Comment by nikic** [comment]
What we could additionally specify are the variance rules of `iterable` (e.g., in a return type you can replace it with `array` or `Traversable`). However, if we don't specify variance for the rest, it's fine as-is.

**Comment by nikic** [comment]
Another thing: In https://github.com/php/php-langspec/blob/master/spec/09-lexical-structure.md#names there is a sentence "The following names cannot be used as the names of classes, interfaces, or traits: bool, FALSE, float, int, NULL, string, TRUE, and void." This should be extended with `iterable`.

**Comment by Hywan** [comment]
Thanks for noticing this.

**Comment by Hywan** [comment]
Good job for `grammar.php` and other tools by the way.

**Comment by nikic** [comment]
Merged via https://github.com/php/php-langspec/commit/248ec2865a64827570a6894c732f5e6b481e0d95. Thanks!

**Comment by Hywan** [comment]
Thanks!

**Pull request closed without merging**